torch/csrc/utils/invalid_arguments.cpp

```diff
@@ -119,36 +125,36 @@ struct Option {
   bool has_out;
 };
 
-std::vector<std::string> _splitString(
-    const std::string& s,
-    const std::string& delim) {
-  std::vector<std::string> tokens;
+std::vector<std::string_view> _splitString(
+    std::string_view s,
+    std::string_view delim) {
+  std::vector<std::string_view> tokens;
   size_t start = 0;
   size_t end = 0;
-  while ((end = s.find(delim, start)) != std::string::npos) {
+  while ((end = s.find(delim, start)) != std::string_view::npos) {
     tokens.push_back(s.substr(start, end - start));
     start = end + delim.length();
   }
   tokens.push_back(s.substr(start));
   return tokens;
 }
 
-std::unique_ptr<Type> _buildType(std::string type_name, bool is_nullable) {
+std::unique_ptr<Type> _buildType(std::string_view type_name, bool is_nullable) {
   std::unique_ptr<Type> result;
   if (type_name == "float") {
     result = torch::make_unique<MultiType>(MultiType{"float", "int", "long"});
   } else if (type_name == "int") {
     result = torch::make_unique<MultiType>(MultiType{"int", "long"});
   } else if (type_name.find("tuple[") == 0) {
     auto type_list = type_name.substr(6);
-    type_list.pop_back();
+    type_list.remove_suffix(1);
     std::vector<std::unique_ptr<Type>> types;
     for (auto& type : _splitString(type_list, ","))
       types.emplace_back(_buildType(type, false));
```

Comment:
While you are touching this file, mind reserving the size of this vector if possible?

Reply:
The CI is green, I will add it in the next patch.

Reply:
Fixed